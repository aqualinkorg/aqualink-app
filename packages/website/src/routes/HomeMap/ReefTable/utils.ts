import { startCase } from "lodash";
import type { TableRow } from "../../../store/Homepage/types";

export type Order = "asc" | "desc";

export enum OrderKeys {
  ALERT = "alert",
  DEPTH = "depth",
  DHW = "dhw",
  LOCATION_NAME = "locationName",
  TEMP = "temp",
}

// This type isn't used anywhere, it just forces the above enum to only hold valid keys to TableRow.
type __CheckOrderKeys = TableRow[OrderKeys];

/**
 * Returns a human friendly string of a order key.
 * @param orderKey key to return the string for.
 */
export function getOrderKeysFriendlyString(orderKey: OrderKeys) {
  switch (orderKey) {
    case OrderKeys.DHW:
      return "DHW";
    default:
      return startCase(orderKey);
  }
}

export function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
  return b[orderBy] >= a[orderBy] ? 1 : -1;
}

export function getComparator(
  order: Order,
  orderBy: OrderKeys
): (
  a: {
    [key in OrderKeys]: number | string | null;
  },
  b: {
    [key in OrderKeys]: number | string | null;
  }
) => number {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

export function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
  // eslint-disable-next-line fp/no-mutating-methods
  return [...array].sort(comparator);
}
