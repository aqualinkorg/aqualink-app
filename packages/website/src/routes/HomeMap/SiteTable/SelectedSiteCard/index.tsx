import React from 'react';
import { Box, Card, Theme, Typography } from '@mui/material';
import LaunchIcon from '@mui/icons-material/Launch';
import { Link } from 'react-router-dom';

import { useSelector } from 'react-redux';
import makeStyles from '@mui/styles/makeStyles';

import {
  siteDetailsSelector,
  siteErrorSelector,
  siteLoadingSelector,
} from 'store/Sites/selectedSiteSlice';
import { surveyListSelector } from 'store/Survey/surveyListSlice';
import { sortByDate } from 'helpers/dates';
import LoadingSkeleton from 'common/LoadingSkeleton';
import SelectedSiteCardContent from './CardContent';

const featuredSiteId = process.env.REACT_APP_FEATURED_SITE_ID || '';

const SelectedSiteCard = () => {
  const classes = useStyles();
  const site = useSelector(siteDetailsSelector);
  const loading = useSelector(siteLoadingSelector);
  const error = useSelector(siteErrorSelector);
  const surveyList = useSelector(surveyListSelector);

  const isFeatured = (site?.id || '').toString() === featuredSiteId;

  const { featuredSurveyMedia } =
    sortByDate(surveyList, 'diveDate', 'desc').find(
      (survey) =>
        survey.featuredSurveyMedia &&
        survey.featuredSurveyMedia.type === 'image',
    ) || {};

  const hasMedia = Boolean(featuredSurveyMedia?.url);

  // If FEATURED_SITE is not setup, no card is displayed.
  if (featuredSiteId === '' && isFeatured) {
    return null;
  }

  return (
    <Box className={classes.card}>
      <Box mb={2}>
        <LoadingSkeleton
          loading={loading}
          variant="text"
          lines={1}
          textHeight={28}
        >
          {site && (
            <Typography variant="h5" color="textSecondary">
              {isFeatured ? 'Featured Site' : 'Selected Site'}
              {!hasMedia && (
                <Link to={`/sites/${site?.id}`}>
                  <LaunchIcon className={classes.launchIcon} />
                </Link>
              )}
            </Typography>
          )}
        </LoadingSkeleton>
      </Box>

      <Card>
        <SelectedSiteCardContent
          site={site}
          loading={loading}
          error={error}
          imageUrl={
            featuredSurveyMedia?.thumbnailUrl || featuredSurveyMedia?.url
          }
        />
      </Card>
    </Box>
  );
};

const useStyles = makeStyles((theme: Theme) => ({
  card: {
    [theme.breakpoints.down('md')]: {
      padding: 10,
    },
    padding: 20,
  },
  launchIcon: {
    fontSize: 20,
    marginLeft: '0.5rem',
    color: '#2f2f2f',
    '&:hover': {
      color: '#2f2f2f',
    },
  },
}));

export default SelectedSiteCard;
