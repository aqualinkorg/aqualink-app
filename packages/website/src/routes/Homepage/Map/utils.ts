export const coloredBuoy = (color: string) =>
  `<?xml version="1.0" encoding="UTF-8"?>
  <svg width="29px" height="35px" viewBox="0 0 29 35" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
      <defs>
          <path d="M10.0023993,-6.25277607e-13 C17.4094117,-6.25277607e-13 22.207967,7.82436546 18.9739831,14.3966715 C17.4094871,17.7347423 12.1924976,22.7424773 10.0023993,25.1417547 C7.70723368,22.5332976 2.80456638,17.9439217 1.13490246,14.5007584 C-2.30725517,8.03253919 2.49117441,-6.25277607e-13 10.0023993,-6.25277607e-13 Z" id="path-1"></path>
          <filter x="-35.0%" y="-19.9%" width="170.0%" height="155.7%" filterUnits="objectBoundingBox" id="filter-2">
              <feOffset dx="0" dy="2" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset>
              <feGaussianBlur stdDeviation="2" in="shadowOffsetOuter1" result="shadowBlurOuter1"></feGaussianBlur>
              <feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0" type="matrix" in="shadowBlurOuter1"></feColorMatrix>
          </filter>
      </defs>
      <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
          <g id="Reef-Detail-full---no-live-feed-Copy-2" transform="translate(-748.000000, -491.000000)">
              <g id="icon_sat" transform="translate(752.500000, 493.500000)">
                  <g id="Fill-1">
                      <use fill="black" fill-opacity="1" filter="url(#filter-2)" xlink:href="#path-1"></use>
                      <use fill="${color}" fill-rule="evenodd" xlink:href="#path-1"></use>
                  </g>
                  <path d="M15.2680598,9.87791599 C11.5105654,13.6354103 6.18310234,8.31883808 9.9461188,4.555975 L12.1649364,6.77540616 L12.1649364,6.41409093 C12.8785627,6.41409093 13.4591827,6.99710385 13.4591827,7.7107302 L13.0996622,7.70953221 L15.2680598,9.87791599 Z M16.5730897,7.82384158 C16.5736894,5.33842841 14.5375756,3.3 12.0515489,3.3 L12.0515489,4.19818779 C14.0402783,4.19818779 15.6772794,5.83150751 15.6772794,7.82085043 L16.5730897,7.82324642 L16.5730897,7.82384158 Z M15.0787125,7.78729129 C15.0787125,6.1436943 13.7305335,4.79553061 12.0869518,4.79553061 L12.0869518,5.69431663 C13.2361966,5.69431663 14.1799878,6.63803121 14.1799878,7.78735265 L15.0787125,7.78729129 Z M14.2254993,14.820015 L13.1571229,15.8883607 C12.6082704,16.4372131 11.7100826,16.4372131 11.1611841,15.8883607 L9.19592376,13.9231004 L11.2625763,11.8564478 L14.2254993,14.8193708 L14.2254993,14.820015 Z M4.90689491,5.50141065 L3.8385495,6.56978704 C3.28969701,7.11863953 3.28969701,8.01682732 3.8385495,8.56572583 L5.80380954,10.5309862 L7.87046211,8.46433361 L4.90753915,5.50141065 L4.90689491,5.50141065 Z M7.7537,13.1817868 L6.98793566,13.9475511 C6.76803266,14.1668559 6.40912568,14.1668559 6.18922268,13.9475511 L5.77877594,13.5371044 C5.55947117,13.3172014 5.55947117,12.9582944 5.77877594,12.7383914 L6.54454028,11.9726271 C6.58228856,12.0427319 6.63082337,12.1086429 6.68954343,12.1673584 L7.55837196,13.0361869 C7.61709201,13.094907 7.68300309,13.1434418 7.75250507,13.1811901 L7.7537,13.1817868 Z M10.5345618,11.2584025 L9.13185186,12.6610971 C8.80230414,12.9906448 8.26362156,12.9906448 7.93407384,12.6610971 L7.06524531,11.7922685 C6.73569759,11.4627208 6.73569759,10.9240382 7.06524531,10.5944905 L8.50748441,9.15225143 C8.86340024,10.0426469 9.65732738,10.8653351 10.5345618,11.2578656 L10.5345618,11.2584025 Z" id="Fill-1" fill="#FFFFFF" transform="translate(10.000000, 9.800000) scale(-1, -1) translate(-10.000000, -9.800000) "></path>
              </g>
          </g>
      </g>
  </svg>`;
