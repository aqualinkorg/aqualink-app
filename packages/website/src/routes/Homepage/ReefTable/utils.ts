export type Order = "asc" | "desc";

export type OrderKeys = "locationName" | "temp" | "depth" | "dhw" | "alert";

export function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
  return b[orderBy] >= a[orderBy] ? 1 : -1;
}

export function getComparator(
  order: Order,
  orderBy: OrderKeys
): (
  a: {
    [key in OrderKeys]: number | string | null;
  },
  b: {
    [key in OrderKeys]: number | string | null;
  }
) => number {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

export function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
  return [...array].sort(comparator);
}
