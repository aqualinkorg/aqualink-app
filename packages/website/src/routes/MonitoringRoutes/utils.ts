import { OptionsObject, SnackbarKey, SnackbarMessage } from 'notistack';
import React from 'react';
import { User } from 'store/User/types';

interface FetchDataProps<T> {
  user: User | null;
  enqueueSnackbar: (
    message: SnackbarMessage,
    options?: OptionsObject | undefined,
  ) => SnackbarKey;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
  setResult?: React.Dispatch<React.SetStateAction<T | null>>;
  getResult: (token: string) => Promise<T>;
}

export async function fetchData<T>({
  user,
  enqueueSnackbar,
  setLoading,
  setResult,
  getResult,
}: FetchDataProps<T>) {
  const { token } = user || {};
  if (!token) {
    enqueueSnackbar('User is not authenticated!', {
      variant: 'error',
    });
    return;
  }

  if (setResult) {
    setResult(null);
  }

  setLoading(true);
  try {
    const data = await getResult(token);

    if (setResult) {
      setResult(data);
    }
  } catch (error: any) {
    const message = error?.response?.data?.message || error?.message;
    const errorMessage =
      typeof message === 'object' ? message?.join('. ') : message;
    enqueueSnackbar(errorMessage || 'Request failed', {
      variant: 'error',
    });
  } finally {
    setLoading(false);
  }
}

export const includes = (
  a: string | undefined | null,
  b: string | undefined,
) => {
  if (!b?.trim()) return true;
  if (!a) return false;

  return a.toLocaleLowerCase().includes(b.toLocaleLowerCase());
};
