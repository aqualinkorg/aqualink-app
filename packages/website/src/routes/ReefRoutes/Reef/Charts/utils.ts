import type { Data } from "../../../../store/Reefs/types";

export const createDatasets = (dailyData: Data[]) => {
  const dailyDataLen = dailyData.length;

  const bottomTemperature = dailyData.map((item) => item.avgBottomTemperature);
  const surfaceTemperature = dailyData
    .filter(
      (item) =>
        item.surfaceTemperature !== null || item.satelliteTemperature !== null
    )
    .map((item) => item.surfaceTemperature || item.satelliteTemperature);

  const meanBottomTemperature =
    bottomTemperature.reduce((a, b) => a + b) / dailyDataLen;
  const meanSurfaceTemperature =
    surfaceTemperature.reduce((a, b) => a + b) / dailyDataLen;

  return {
    bottomTemperatureData: [
      meanBottomTemperature,
      ...bottomTemperature,
      meanBottomTemperature,
    ],
    surfaceTemperatureData: [
      meanSurfaceTemperature,
      ...surfaceTemperature,
      meanSurfaceTemperature,
    ],
  };
};

export const calculateAxisLimits = (dailyData: Data[]) => {
  const dates = dailyData
    .filter(
      (item) =>
        item.surfaceTemperature !== null || item.satelliteTemperature !== null
    )
    .map((item) => item.date);
  const dailyDataLen = dates.length;

  const xAxisMax = new Date(
    new Date(dates[dailyDataLen - 1]).setHours(24, 0, 0, 0)
  ).toISOString();

  const xAxisMin = new Date(
    new Date(dates[0]).setHours(-1, 0, 0, 0)
  ).toISOString();

  // Add an extra date one day after the final daily data date
  const chartLabels = [
    new Date(new Date(xAxisMin).setHours(3, 0, 0, 0)).toISOString(),
    ...dates,
    new Date(new Date(xAxisMax).setHours(3, 0, 0, 0)).toISOString(),
  ];

  return {
    xAxisMax,
    xAxisMin,
    chartLabels,
  };
};
