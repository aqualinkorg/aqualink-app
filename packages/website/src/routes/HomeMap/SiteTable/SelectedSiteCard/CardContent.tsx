import React from 'react';
import {
  Theme,
  useTheme,
  useMediaQuery,
  Grid,
  Box,
  CardMedia,
  Hidden,
  Typography,
  Button,
  Alert,
} from '@mui/material';
import makeStyles from '@mui/styles/makeStyles';
import { isNumber } from 'lodash';
import { Link, useLocation } from 'react-router-dom';
import classNames from 'classnames';

import { Site } from 'store/Sites/types';
import { getSiteNameAndRegion } from 'store/Sites/helpers';
import { formatNumber } from 'helpers/numberUtils';
import Chart from 'common/Chart';
import { standardDailyDataDataset } from 'common/Chart/MultipleSensorsCharts/helpers';
import Chip from 'common/Chip';
import LoadingSkeleton from 'common/LoadingSkeleton';
import { GaAction, GaCategory, trackButtonClick } from 'utils/google-analytics';
import featuredImageLoading from '../../../../assets/img/loading-image.svg';
import chartLoading from '../../../../assets/img/chart_skeleton.png';

const SelectedSiteCardContent = ({
  site,
  loading,
  error,
  imageUrl = null,
}: SelectedSiteCardContentProps) => {
  const classes = useStyles({ imageUrl, loading });
  const theme = useTheme();
  const location = useLocation();
  const isTablet = useMediaQuery(theme.breakpoints.down('md'));
  const {
    bottomTemperature,
    satelliteTemperature,
    dhw: degreeHeatingWeek,
  } = site?.collectionData || {};
  const useCardWithImageLayout = Boolean(loading || imageUrl);

  const metrics = [
    {
      label: 'SURFACE TEMP',
      value: formatNumber(satelliteTemperature, 1),
      unit: ' °C',
      display: isNumber(satelliteTemperature),
    },
    {
      label: 'HEAT STRESS',
      value: formatNumber(degreeHeatingWeek, 1),
      unit: ' DHW',
      display: isNumber(degreeHeatingWeek),
    },
    {
      label: `TEMP AT ${site?.depth}m`,
      value: formatNumber(bottomTemperature, 1),
      unit: ' °C',
      display: isNumber(bottomTemperature) && isNumber(site?.depth),
    },
  ];

  const { name, region: regionName } = site
    ? getSiteNameAndRegion(site)
    : { name: null, region: null };

  const chartDataset = site
    ? standardDailyDataDataset(
        site.dailyData,
        site.maxMonthlyMean,
        false,
        site.timezone,
      )
    : undefined;

  const ChartComponent =
    site && chartDataset ? (
      <Chart
        siteId={site.id}
        datasets={[chartDataset]}
        surveys={[]}
        temperatureThreshold={
          site.maxMonthlyMean ? site.maxMonthlyMean + 1 : null
        }
        maxMonthlyMean={site.maxMonthlyMean || null}
        background
      />
    ) : null;

  const onExploreButtonClick = () => {
    trackButtonClick(
      GaCategory.BUTTON_CLICK,
      GaAction.MAP_PAGE_BUTTON_CLICK,
      'Explore',
    );
  };

  if (error) {
    return <Alert severity="error">{error}</Alert>;
  }

  if (!loading && !site) {
    return <Alert severity="error">Featured site is not available</Alert>;
  }

  return (
    <Grid
      className={classes.cardWrapper}
      container
      justifyContent="space-between"
      spacing={1}
    >
      {useCardWithImageLayout && (
        <Grid item xs={12} md={6} lg={4}>
          <Box position="relative" height="100%" minHeight={300}>
            <LoadingSkeleton
              className={classes.imageBorderRadius}
              image={featuredImageLoading}
              loading={loading}
              variant="rectangular"
              height="100%"
            >
              {site && imageUrl && (
                <Link
                  to={`/sites/${site.id}`}
                  state={{ from: location.pathname }}
                >
                  <CardMedia
                    className={classNames(
                      classes.cardImage,
                      classes.imageBorderRadius,
                    )}
                    image={imageUrl}
                  />
                </Link>
              )}

              <Hidden mdUp>
                <Box
                  bgcolor="rgba(3, 48, 66, 0.75)"
                  height="55%"
                  width="100%"
                  position="absolute"
                  top={0}
                  left={0}
                >
                  <Box className={classes.cardImageTextWrapper}>
                    <Typography variant="h5" color="white">
                      {name}
                    </Typography>

                    {regionName && (
                      <Typography variant="h6" color="white">
                        {regionName}
                      </Typography>
                    )}
                  </Box>
                </Box>
              </Hidden>

              {site && (
                <Box position="absolute" bottom={16} px={2} width="100%">
                  <Grid
                    container
                    alignItems="center"
                    justifyContent={
                      site.videoStream && isTablet
                        ? 'space-between'
                        : 'flex-end'
                    }
                  >
                    {site.videoStream && isTablet && (
                      <Chip
                        live
                        liveText="LIVE VIDEO"
                        to={`/sites/${site.id}`}
                        state={{ from: location.pathname }}
                        width={80}
                      />
                    )}
                    <Grid item>
                      <Button
                        className={classes.exploreButton}
                        component={Link}
                        to={`/sites/${site.id}`}
                        state={{ from: location.pathname }}
                        onClick={onExploreButtonClick}
                        size="small"
                        variant="contained"
                        color="primary"
                      >
                        EXPLORE
                      </Button>
                    </Grid>
                  </Grid>
                </Box>
              )}
            </LoadingSkeleton>
          </Box>
        </Grid>
      )}
      <Grid
        item
        xs={12}
        md={useCardWithImageLayout ? 6 : 12}
        lg={useCardWithImageLayout ? 6 : 10}
        className={classes.cardAnalyticsWrapper}
      >
        <Box pb="0.5rem" pl="0.5rem" pt="1.5rem" fontWeight={400}>
          <Hidden mdDown={useCardWithImageLayout}>
            <LoadingSkeleton
              loading={loading}
              variant="text"
              lines={1}
              textHeight={28}
              longText
            >
              <Grid container alignItems="center">
                <Grid item className={classes.cardTitleWrapper}>
                  <Typography
                    className={classes.cardTitle}
                    color="textSecondary"
                    variant="h5"
                  >
                    <span title={name || ''}>{name}</span>
                  </Typography>
                </Grid>
                {site?.videoStream && (
                  <Chip
                    live
                    liveText="LIVE VIDEO"
                    to={`/sites/${site.id}`}
                    state={{ from: location.pathname }}
                    width={80}
                  />
                )}
              </Grid>
            </LoadingSkeleton>
            <LoadingSkeleton
              loading={loading}
              variant="text"
              lines={1}
              textHeight={19}
              className={classes.siteRegionName}
            >
              <Typography
                color="textSecondary"
                variant="h6"
                className={classNames(
                  classes.cardTitle,
                  classes.siteRegionName,
                )}
              >
                <span title={regionName || ''}>{regionName}</span>
              </Typography>
            </LoadingSkeleton>
          </Hidden>
          <LoadingSkeleton
            loading={loading}
            variant="text"
            lines={1}
            textHeight={12}
          >
            <Typography color="textSecondary" variant="caption">
              DAILY SURFACE TEMP. (°C)
            </Typography>
          </LoadingSkeleton>
        </Box>
        <Hidden mdDown>
          <LoadingSkeleton
            image={chartLoading}
            loading={loading}
            variant="rectangular"
            width="98%"
            height={180}
          >
            <div>{ChartComponent}</div>
          </LoadingSkeleton>
        </Hidden>
        <Hidden mdUp>
          <LoadingSkeleton
            className={classes.mobileChartLoading}
            image={chartLoading}
            loading={loading}
            variant="rectangular"
            width="98%"
            height={200}
          >
            {ChartComponent}
          </LoadingSkeleton>
        </Hidden>
      </Grid>
      <Grid item xs={12} lg={2} container>
        <div className={classes.metricsContainer}>
          {metrics.map(({ label, value, unit, display }) => (
            <div key={label} className={classes.metric}>
              <LoadingSkeleton
                longText
                loading={loading}
                variant="text"
                lines={1}
                textHeight={12}
              >
                {display && (
                  <Typography variant="caption" color="textSecondary">
                    {label}
                  </Typography>
                )}
              </LoadingSkeleton>
              <LoadingSkeleton
                loading={loading}
                variant="rectangular"
                width={80}
                height={32}
              >
                {display && (
                  <Typography variant="h4" color="primary">
                    {value}
                    &nbsp;
                    <Typography variant="h6" component="span">
                      {unit}
                    </Typography>
                  </Typography>
                )}
              </LoadingSkeleton>
            </div>
          ))}
        </div>
      </Grid>
    </Grid>
  );
};

const useStyles = makeStyles((theme: Theme) => ({
  cardWrapper: ({ imageUrl, loading }: SelectedSiteCardContentStyleProps) => ({
    minHeight: '18rem',
    [theme.breakpoints.down('lg')]: {
      minHeight: '24rem',
    },
    [theme.breakpoints.down('md')]: {
      height: imageUrl || loading ? '42rem' : '27rem',
    },
  }),
  imageBorderRadius: {
    borderRadius: '4px 0 0 4px',
    [theme.breakpoints.down('lg')]: {
      borderRadius: '4px 0 0 0',
    },
    [theme.breakpoints.down('md')]: {
      borderRadius: '4px 4px 0 0',
    },
  },
  cardImage: {
    height: '100%',

    [theme.breakpoints.down('md')]: {
      height: 300,
    },
  },
  cardImageTextWrapper: {
    position: 'absolute',
    top: 16,
    left: 16,
    right: 16,
    overflowWrap: 'break-word',
  },
  cardAnalyticsWrapper: {
    marginBottom: '2rem',
    maxHeight: '14rem',
  },
  cardTitleWrapper: {
    maxWidth: 'calc(100% - 88px)',
    marginRight: theme.spacing(1),
  },
  cardTitle: {
    width: '100%',
    display: 'block',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
  },
  siteRegionName: {
    marginBottom: '0.6rem',
  },
  metricsContainer: {
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
    flexDirection: 'row',
    width: '100%',
    [theme.breakpoints.down('md')]: {
      marginTop: theme.spacing(3),
      marginBottom: theme.spacing(3),
    },

    [theme.breakpoints.up('lg')]: {
      flexDirection: 'column',
      alignItems: 'start',
    },
  },
  metric: {
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  mobileChartLoading: {
    margin: 'auto',
  },
  exploreButton: {
    '&:hover': {
      color: 'white',
    },
  },
}));

interface SelectedSiteCardContentProps {
  site?: Site | null;
  loading: boolean;
  error?: string | null;
  imageUrl?: string | null;
}

type SelectedSiteCardContentStyleProps = Pick<
  SelectedSiteCardContentProps,
  'imageUrl' | 'loading'
>;

export default SelectedSiteCardContent;
