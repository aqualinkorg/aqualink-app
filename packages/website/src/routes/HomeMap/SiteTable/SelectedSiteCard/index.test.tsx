import React from 'react';
import configureStore from 'redux-mock-store';

import { mockSite } from 'mocks/mockSite';
import { renderWithProviders } from 'utils/test-utils';
import SelectedSiteCard from '.';

const site = {
  details: mockSite,
};

const mockStore = configureStore([]);

const store = mockStore({
  selectedSite: site,
  homepage: {
    siteOnMap: site,
  },
  surveyList: {
    list: [],
  },
  survey: {
    selectedSurvey: {
      details: null,
    },
  },
});

store.dispatch = vi.fn();

test('renders as expected', () => {
  process.env.REACT_APP_FEATURED_SITE_ID = '2';

  const { container } = renderWithProviders(<SelectedSiteCard />, { store });
  expect(container).toMatchSnapshot();
});

test('renders loading as expected', () => {
  process.env.REACT_APP_FEATURED_SITE_ID = '4';
  const { container } = renderWithProviders(<SelectedSiteCard />, { store });
  expect(container).toMatchSnapshot();
});
