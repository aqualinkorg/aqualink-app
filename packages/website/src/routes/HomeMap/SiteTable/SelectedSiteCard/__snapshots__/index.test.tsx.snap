// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-box
    classname="makeStyles-card-1"
  >
    <mock-box
      mb="2"
    >
      <mock-typography
        color="textSecondary"
        variant="h5"
      >
        Selected Site
        <a
          href="/sites/1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-launchIcon-2 css-1umw9bq-MuiSvgIcon-root"
            data-testid="LaunchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
            />
          </svg>
        </a>
      </mock-typography>
    </mock-box>
    <mock-card>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 makeStyles-cardWrapper-9 makeStyles-cardWrapper-21 css-1h90obp-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-10 makeStyles-cardAnalyticsWrapper-13 css-1nyv9pn-MuiGrid-root"
        >
          <mock-box
            fontweight="400"
            pb="0.5rem"
            pl="0.5rem"
            pt="1.5rem"
          >
            <mock-hidden
              mddown="false"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-1rp4aer-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item makeStyles-cardTitleWrapper-14 css-x7bhrq-MuiGrid-root"
                >
                  <mock-typography
                    classname="makeStyles-cardTitle-15"
                    color="textSecondary"
                    variant="h5"
                  >
                    <span
                      title="Hawaii"
                    >
                      Hawaii
                    </span>
                  </mock-typography>
                </div>
              </div>
              <mock-typography
                classname="makeStyles-cardTitle-15 makeStyles-siteRegionName-16"
                color="textSecondary"
                variant="h6"
              >
                <span
                  title=""
                />
              </mock-typography>
            </mock-hidden>
            <mock-typography
              color="textSecondary"
              variant="caption"
            >
              DAILY SURFACE TEMP. (°C)
            </mock-typography>
          </mock-box>
          <mock-hidden
            mddown="true"
          >
            <div>
              <mock-line
                data="[object Object]"
                options="[object Object]"
              />
            </div>
          </mock-hidden>
          <mock-hidden
            mdup="true"
          >
            <mock-line
              data="[object Object]"
              options="[object Object]"
            />
          </mock-hidden>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-2 css-15vsh4j-MuiGrid-root"
        >
          <div
            class="makeStyles-metricsContainer-17"
          >
            <div
              class="makeStyles-metric-18"
            >
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                SURFACE TEMP
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                15.9
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
            <div
              class="makeStyles-metric-18"
            >
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                HEAT STRESS
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                0.0
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   DHW
                </mock-typography>
              </mock-typography>
            </div>
            <div
              class="makeStyles-metric-18"
            >
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                TEMP AT 0m
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                10.7
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
          </div>
        </div>
      </div>
    </mock-card>
  </mock-box>
</div>
`;

exports[`renders loading as expected 1`] = `
<div>
  <mock-box
    classname="makeStyles-card-44"
  >
    <mock-box
      mb="2"
    >
      <mock-typography
        color="textSecondary"
        variant="h5"
      >
        Selected Site
        <a
          href="/sites/1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-launchIcon-45 css-1umw9bq-MuiSvgIcon-root"
            data-testid="LaunchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z"
            />
          </svg>
        </a>
      </mock-typography>
    </mock-box>
    <mock-card>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 makeStyles-cardWrapper-52 makeStyles-cardWrapper-64 css-1h90obp-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-10 makeStyles-cardAnalyticsWrapper-56 css-1nyv9pn-MuiGrid-root"
        >
          <mock-box
            fontweight="400"
            pb="0.5rem"
            pl="0.5rem"
            pt="1.5rem"
          >
            <mock-hidden
              mddown="false"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-1rp4aer-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item makeStyles-cardTitleWrapper-57 css-x7bhrq-MuiGrid-root"
                >
                  <mock-typography
                    classname="makeStyles-cardTitle-58"
                    color="textSecondary"
                    variant="h5"
                  >
                    <span
                      title="Hawaii"
                    >
                      Hawaii
                    </span>
                  </mock-typography>
                </div>
              </div>
              <mock-typography
                classname="makeStyles-cardTitle-58 makeStyles-siteRegionName-59"
                color="textSecondary"
                variant="h6"
              >
                <span
                  title=""
                />
              </mock-typography>
            </mock-hidden>
            <mock-typography
              color="textSecondary"
              variant="caption"
            >
              DAILY SURFACE TEMP. (°C)
            </mock-typography>
          </mock-box>
          <mock-hidden
            mddown="true"
          >
            <div>
              <mock-line
                data="[object Object]"
                options="[object Object]"
              />
            </div>
          </mock-hidden>
          <mock-hidden
            mdup="true"
          >
            <mock-line
              data="[object Object]"
              options="[object Object]"
            />
          </mock-hidden>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-2 css-15vsh4j-MuiGrid-root"
        >
          <div
            class="makeStyles-metricsContainer-60"
          >
            <div
              class="makeStyles-metric-61"
            >
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                SURFACE TEMP
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                15.9
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
            <div
              class="makeStyles-metric-61"
            >
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                HEAT STRESS
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                0.0
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   DHW
                </mock-typography>
              </mock-typography>
            </div>
            <div
              class="makeStyles-metric-61"
            >
              <mock-typography
                color="textSecondary"
                variant="caption"
              >
                TEMP AT 0m
              </mock-typography>
              <mock-typography
                color="primary"
                variant="h4"
              >
                10.7
                 
                <mock-typography
                  component="span"
                  variant="h6"
                >
                   °C
                </mock-typography>
              </mock-typography>
            </div>
          </div>
        </div>
      </div>
    </mock-card>
  </mock-box>
</div>
`;
